### PR TITLE
[Issue #10] Ensure words do not shuffle in the original order

### PR DESCRIPTION
**Pull request by cbouwense**
Implementation of #10 

## TL;DR 
If a word gets shuffled into its original order, just shuffle it manually.

## Long Version
- Implement and test a util function for checking if arrays are equal. This is used for checking if the shuffled letters are in the original order.
- Break up the shuffle module into a few smaller functions. The default export's logic is now a wrapper for the original shuffling behavior. However, it does the checking of the order of the word shuffling.
- Extract getting a random number generator to an injector. This enables us to mock what the rng will produce, and thus reproduce the conditions in which the order is unchanged.

### Miscellaneous
I also enabled strict null checks, because it occurred to me that this codebase did not already have it on.

### CODE REVIEW

File: src/lib/utils/areArraysEqual.test.ts [hunk 1]
```diff
@@ -0,0 +1,37 @@
...
+	it('returns true', () => {
+		const arr1 = [1, 2, 3];
+		const arr2 = [1, 2, 3];
+
+		expect(areArraysEqual(arr1, arr2)).toBe(true);
+	});
+});
+
+describe('given arrays with different elements', () => {
+	it('returns true', () => {
```
Comment (cbouwense) [inline src/lib/utils/areArraysEqual.test.ts:13]:
```suggestion
	it('returns false', () => {
```

File: src/lib/utils/areArraysEqual.test.ts [hunk 2]
```diff
@@ -0,0 +1,37 @@
...
+		const arr1 = [1, 2, 3];
+		const arr2 = [1, 3, 3, 7];
+
+		expect(areArraysEqual(arr1, arr2)).toBe(false);
+	});
+});
+
+describe('given arrays with different types of elements', () => {
+	it('returns true', () => {
```
Comment (cbouwense) [inline src/lib/utils/areArraysEqual.test.ts:22]:
```suggestion
	it('returns false', () => {
```

File: src/lib/utils/shuffle.ts [hunk 3]
```diff
@@ -1,14 +1,32 @@
...
-export default (array: string[]) => {
-	const today = new Date();
-	const rng = new Prando(today.toDateString() + get(gameProgress));
-	for (let i = array.length - 1; i > 0; i--) {
-		const j = rng.nextInt(0, i + 1);
-		[array[i], array[j]] = [array[j], array[i]];
+import areArraysEqual from './areArraysEqual';
+import rng from './getRandomNumberGenerator';
+
+export const getShuffledWord = (letters: string[]): string[] => {
```
Comment (cbouwense) [inline src/lib/utils/shuffle.ts:4]: I used the term "word" a lot in this file but now I'm realizing that these aren't really words we're scrambling, I should probably rename these

File: src/lib/utils/shuffle.test.ts [hunk 4]
```diff
@@ -0,0 +1,32 @@
+import { vi, type Mock } from 'vitest';
+import rng from './getRandomNumberGenerator';
+import getShuffledWord, { rearrangeLettersManually } from './shuffle';
+
+// TODO: Figure out how to not mock these store modules.
+vi.mock('svelte/store', () => ({ get: vi.fn() }));
+vi.mock('$lib/store', () => ({ gameProgress: 0 }));
+vi.mock('./getRandomNumberGenerator');
```
[the remark this answers is not in the document]
Comment (AlexMassenzio) [reply src/lib/utils/shuffle.test.ts:8]: Yeah we should add store mocking to setupTests possibly

**Review by AlexMassenzio — COMMENTED** [review]
I should look at this on Desktop. Will be able to tomorrow most likely.

File: src/lib/utils/shuffle.test.ts (hunk 4 above)
[the remark this answers is not in the document]
Comment (AlexMassenzio) [reply src/lib/utils/shuffle.test.ts:8]: #18

**Comment by AlexMassenzio** [comment]
We should deploy this at around midnight.

**Comment by AlexMassenzio** [comment]
Lets deploy this at midnight to not affect any of the current day's submissions.

I'm also going to change the branch to merge to staging to bundle multiple changes in one for tonight.

**Review by AlexMassenzio — APPROVED** [review]
Cleared for merge.
